compiler_opt/rl/compilation_runner.py

```diff
@@ -48,6 +49,38 @@ class RewardStat:
   moving_average_reward: float
 
 
+@dataclasses.dataclass(frozen=True)
+class Policy:
+  """Serialized mlgo policy, used to pass a policy to workers.
+
+  A policy has 2 components, both being file contents:
+    - the content of the output_spec.json file;
+    - the content of the tflite policy.
+  """
+
+  output_spec: bytes
+  policy: bytes
+
+  def to_filesystem(self, location: str):
+    os.makedirs(location, exist_ok=True)
+    output_sig = os.path.join(location, policy_saver.OUTPUT_SIGNATURE)
+    policy_path = os.path.join(location, policy_saver.TFLITE_MODEL_NAME)
+    with tf.io.gfile.GFile(output_sig, mode='wb') as f:
+      f.write(self.output_spec)
+    with tf.io.gfile.GFile(policy_path, mode='wb') as f:
+      f.write(self.policy)
+
+  @staticmethod
+  def from_filesystem(location: str):
+    output_sig = os.path.join(location, policy_saver.OUTPUT_SIGNATURE)
+    policy_path = os.path.join(location, policy_saver.TFLITE_MODEL_NAME)
+    with tf.io.gfile.GFile(output_sig, mode='rb') as f:
+      output_spec = f.read()
+    with tf.io.gfile.GFile(policy_path, mode='rb') as f:
+      policy = f.read()
+    return Policy(output_spec=output_spec, policy=policy)
```

Comment:
do you want to define an object or just a bytes (like what we are doing today) for it? the Policy object adds more additional complexity when you passing it through rpcs in the distributed case

Reply:
What complexity - there's no difference between how this object is sent in the local-multiprocess case or distributed case.

Reply:
How does the trainer and the server talk with each other in the new distributed case? not in RPCs?

Reply:
That's handled transparently. The objects, either way (inter-process, local or distributed) are serialized/deserialized the same way under the hood.

Reply:
i mean is the new distributed framework able to handle the serialization/deserialization of a python object automatically? In our current grpc setup, it cannot --- training in python and data collection in C++, even if the data collection is in python, we need to do the serialization/deserialization of the python object ourselves then we can pass it through rpc request

Reply:
The short answer is yes - that's part of the value, we don't need to deal with any of that gRPC, protobufs, any of that. Any pickleable object works.

Basically all the code currently written doesn't need to change. We just need to de-localize the source files (the module .bc/thinlto - a subsequent patch) and that's it.